File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.py
class Solution(object):
    def maxAscendingSum(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        result=nums[0]
        
        temp_Sum=nums[0]

        for i in range(1,len(nums)):
            if(nums[i]>nums[i-1]):
                temp_Sum+=nums[i]
            else:
                temp_Sum=nums[i]
            result=max(temp_Sum,result)
        return result
        

File: 1927-maximum-ascending-subarray-sum/maximum-ascending-subarray-sum.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        
        int result=nums[0];
        int temp_sum=nums[0];
        for(int i=1;i<nums.size();i++){

            if(nums[i]>nums[i-1]){
                temp_sum+=nums[i];
            }else{
                temp_sum=nums[i];
            }
            result=max(temp_sum,result);
        }
        return result;
    }
};